src/libraries/Microsoft.Extensions.Logging/src/LoggerFactory.cs

```diff
@@ -62,8 +62,21 @@ public LoggerFactory(IEnumerable<ILoggerProvider> providers, IOptionsMonitor<Log
         /// <param name="providers">The providers to use in producing <see cref="ILogger"/> instances.</param>
         /// <param name="filterOption">The filter option to use.</param>
         /// <param name="options">The <see cref="LoggerFactoryOptions"/>.</param>
-        public LoggerFactory(IEnumerable<ILoggerProvider> providers, IOptionsMonitor<LoggerFilterOptions> filterOption, IOptions<LoggerFactoryOptions>? options = null)
+        public LoggerFactory(IEnumerable<ILoggerProvider> providers, IOptionsMonitor<LoggerFilterOptions> filterOption, IOptions<LoggerFactoryOptions>? options) : this(providers, filterOption, options, null)
         {
+        }
+
+        /// <summary>
+        /// Creates a new <see cref="LoggerFactory"/> instance.
+        /// </summary>
+        /// <param name="providers">The providers to use in producing <see cref="ILogger"/> instances.</param>
+        /// <param name="filterOption">The filter option to use.</param>
+        /// <param name="options">The <see cref="LoggerFactoryOptions"/>.</param>
+        /// <param name="scopeProvider">The <see cref="IExternalScopeProvider"/>.</param>
```

Comment:
Should this document that LoggerFactoryOptions.ActivityTrackingOptions takes effect only if `scopeProvider` is null?

Reply:
I think it's good to keep validating LoggerFactoryOptions.ActivityTrackingOptions even if `scopeProvider` is not null.

Reply:
> Should this document that LoggerFactoryOptions.ActivityTrackingOptions takes effect only if `scopeProvider` is null?

I see your point. However, I also think it should be safe enough to assume someone who is willing to provide a custom IExternalScopeProvider, wants to do so because they want to manage everything, including trace context.

Reply:
> I think it's good to keep validating LoggerFactoryOptions.ActivityTrackingOptions even if `scopeProvider` is not null.

If `scopeProvider` is not null, `LoggerFactoryOptions.ActivityTrackingOptions` will not be used. Therefore, validating something that is unused and will have no effect seems redundant.

Reply:
Do you intend to make the validation conditional, then?

I expect that changing the validation to `if (scopeProvider is null && (_factoryOptions.ActivityTrackingOptions & ActivityTrackingOptionsMask) != 0)` would just make it slower without any benefit to application developers.

Adding `&& scopeProvider is null` to the

Reply:
Sorry, I misread your comment, I thought you were suggesting adding validation. Although it is redundant validation when `scopeProvider` is null, the argument for keeping it is compelling also: invalid options will be caught even when options are not relevant. Also, it could be risky if future changes cause this assumption to no longer be true, the conditional part of the check could be missed. So, I am in favor of keeping the check.